src/components/policy/policy_external/test/sql_pt_ext_representation_storage_test.cc

```diff
@@ -0,0 +1,124 @@
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "gtest/gtest.h"
+#include "policy/sql_pt_ext_representation.h"
+#include "policy/mock_policy_settings.h"
+#include "utils/shared_ptr.h"
+#include "utils/make_shared.h"
+#include "utils/file_system.h"
+#include "sqlite_wrapper/sql_query.h"
+
+#include <string>
+
+using namespace ::policy;
+
+using testing::ReturnRef;
+using testing::NiceMock;
+
+namespace test {
+namespace components {
+namespace policy_test {
+
+class SQLPTExtRepresentationStorageTest : public ::testing::Test {
+ public:
+  SQLPTExtRepresentationStorageTest() : reps_(NULL) {}
+
+ protected:
+  const std::string kAppStorageFolder =
+      "storage_SQLPTExtRepresentationStorageTest";
+
+  std::shared_ptr<SQLPTExtRepresentation> reps_;
+  std::shared_ptr<NiceMock<policy_handler_test::MockPolicySettings> >
+      policy_settings_;
+  std::shared_ptr<utils::dbms::SQLQuery> query_wrapper_;
+
+  void SetUp() OVERRIDE {
+    file_system::CreateDirectory(kAppStorageFolder);
+
+    reps_ = std::make_shared<SQLPTExtRepresentation>();
+    ASSERT_TRUE(reps_ != NULL);
+
+    policy_settings_ =
+        std::make_shared<NiceMock<policy_handler_test::MockPolicySettings> >();
+    ON_CALL(*policy_settings_, app_storage_folder())
+        .WillByDefault(ReturnRef(kAppStorageFolder));
+    ASSERT_EQ(SUCCESS, reps_->Init(policy_settings_.get()));
+
+    query_wrapper_ = std::make_shared<utils::dbms::SQLQuery>(reps_->db());
+    ASSERT_TRUE(query_wrapper_ != NULL);
+  }
+
+  void TearDown() OVERRIDE {
+    EXPECT_TRUE(reps_->Drop());
+    EXPECT_TRUE(reps_->Close());
+
+    query_wrapper_.reset();
+    reps_.reset();
+    policy_settings_.reset();
+
+    file_system::remove_directory_content(kAppStorageFolder);
+    file_system::RemoveDirectory(kAppStorageFolder, true);
+  }
+};
+
+TEST_F(
+    SQLPTExtRepresentationStorageTest,
+    CanAppKeepContext_InsertKeepContext_ExpectValuesThatSetInKeepContextParams) {
+  // Arrange
+  const std::string query_delete = "DELETE FROM `application`; ";
+
+  // Assert
+  ASSERT_TRUE(query_wrapper_->Exec(query_delete));
+
+  // Act
+  const std::string query_insert_12345 =
+      "INSERT INTO `application` (`id`, `memory_kb`,"
+      " `heart_beat_timeout_ms`, `keep_context`) VALUES ('12345', 5, 10, 1)";
+
+  // Assert
+  ASSERT_TRUE(query_wrapper_->Exec(query_insert_12345));
+  EXPECT_FALSE(reps_->CanAppKeepContext("0"));
+  EXPECT_TRUE(reps_->CanAppKeepContext("12345"));
```

Comment:
@ZhdanovP please move this value to the separate constant and pass it to `query_insert_12345` and check in `CanAppKeepContext` to be sure that you are using the same value